app/pages/lab/mobile-section.jsx

```diff
@@ -0,0 +1,178 @@
+  constructor(props) {
+    super(props);
+    this.toggleSwipeEnabled = this.toggleSwipeEnabled.bind(this);
+  }
+
+  toggleSwipeEnabled({ target }) {
+    const changes = {};
+    changes['configuration.swipe_enabled'] = target.checked;
+    this.props.workflow.update(changes).save();
+    if (target.checked) {
+      this.props.project.update({ mobile_friendly: target.checked }).save();
+    }
+  }
+
+  render() {
+    if (['single', 'multiple'].includes(this.props.task.type) === false) {
+      return null;
+    }
+
+    const nonShortCutTasks = Object.keys(this.props.workflow.tasks)
+      .filter(key => this.props.workflow.tasks[key].type !== 'shortcut');
+    const shortcut = this.props.workflow.tasks[this.props.task.unlinkedTask];
+    const config = this.props.workflow.configuration;
+
+    // the following are criteria for the swipe app
+    const hasTwoAnswers = this.props.task.answers.length === 2;
+    const hasSingleTask = nonShortCutTasks.length === 1;
+    const questionNotTooLong = this.props.task.question.length < 200;
+    const notTooManyShortcuts = shortcut ? shortcut.answers.length <= 2 : true;
+    const isNotFlipbook = config ? config.multi_image_mode !== 'flipbook' : true;
+    const doesNotUseFeedback = this.props.task.feedback ? !this.props.task.feedback.enabled : true;
+
+    const swipeEligible = [
+      hasTwoAnswers,
+      hasSingleTask,
+      questionNotTooLong,
+      notTooManyShortcuts,
+      isNotFlipbook,
+      doesNotUseFeedback
+    ].every(requirement => requirement);
+
+    const ineligibleText = (
+      <small className="form-help">
+        Sorry, but the mobile app will not currently work for this workflow.
+        <br />
+        The following are the requirements for the swipe workflow.
+        <ul>
+          <li>Has one Task { hasSingleTask ? meets : doesntMeet }</li>
+          <li>Has two available answers (e.g. Yes/No) { hasTwoAnswers ? meets : doesntMeet }</li>
+          <li>Question has less than 200 characters { questionNotTooLong ? meets : doesntMeet }</li>
+          <li>Has less than three shortcuts { notTooManyShortcuts ? meets : doesntMeet }</li>
+          <li>Cannot be a flipbook { isNotFlipbook ? meets : doesntMeet }</li>
+          <li>Cannot provide feedback { doesNotUseFeedback ? meets : doesntMeet }</li>
+        </ul>
+      </small>
+    );
+
+    const swipeEnabledChecked = !!this.props.workflow.configuration.swipe_enabled && swipeEligible;
+
+    const launchApproved = this.props.project.launch_approved;
+
+    return (
+      <div>
+        <span className="form-label">Mobile App</span>
+        <br />
+        <div className="workflow-mobile-form">
+          <div className="workflow-mobile-form-left-panel">
+            <label
+              className="pill-button"
+              htmlFor="swipe_enabled"
+              title={mobileHelp}
+              style={(swipeEligible && launchApproved) ? null : disabledStyle}
```

Comment:
Oh nice. My thought was that this would have to be a change on the mobile repo, but this seems to work well also. In mobile, I'll prob also add `launch_approved` to the api query once we have enough fixes/patches to warrant an update.